crates/uv-resolver/src/pubgrub/priority.rs

```diff
@@ -35,25 +36,44 @@ impl PubGrubPriorities {
         version: &Range<Version>,
         urls: &ForkUrls,
     ) {
-        if !self.virtual_package_tiebreaker.contains_key(package) {
-            self.virtual_package_tiebreaker.insert(
-                package.clone(),
-                PubGrubTiebreaker::from(
-                    u32::try_from(self.virtual_package_tiebreaker.len())
-                        .expect("Less than 2**32 packages"),
-                ),
-            );
+        // Pre-compute the key and value.
+        let hash = self.virtual_package_tiebreaker.hasher().hash_one(package);
+        let next = self.virtual_package_tiebreaker.len();
+
+        // Insert into the tiebreaker map, if it doesn't exist.
+        match self
+            .virtual_package_tiebreaker
+            .raw_entry_mut()
+            .from_key_hashed_nocheck(hash, package)
+        {
+            RawEntryMut::Vacant(entry) => {
+                entry.insert_hashed_nocheck(
+                    hash,
+                    package.clone(),
+                    PubGrubTiebreaker::from(u32::try_from(next).expect("Less than 2**32 packages")),
+                );
+            }
+            RawEntryMut::Occupied(_) => {
+                // Nothing to do.
+            }
```

Comment:
What about using the `EntryRef` API?

```rust
        let len = self.virtual_package_tiebreaker.len();
        self.virtual_package_tiebreaker
            .entry_ref(package)
            .or_insert_with(|| {
                PubGrubTiebreaker::from(u32::try_from(len).expect("Less than 2**32 packages"))
            });
```

Reply:
A bit annoying that they use `From` instead of `ToOwned`, but treating it as clone works 

> The relationships I use between the key `K` and borrowed key `Q` is `K: Borrow<Q> + From<&Q>`. The reason for not using `Q: ToOwned<K>` is to support key types like `Rc<str>` (which is used in a couple of the doctests), which would not be possible with `to_owned()`.

(from https://github.com/rust-lang/hashbrown/pull/301)

```rust
impl From<&PubGrubPackage> for PubGrubPackage {
    fn from(package: &PubGrubPackage) -> Self {
        package.clone()
    }
}
```

Reply:
Ok, we can do that instead.